ledger/heavy/handler/handler.go

```diff
@@ -364,24 +365,32 @@ func (h *Handler) handleHeavyPayload(ctx context.Context, genericMsg insolar.Par
 		return &reply.HeavyError{Message: err.Error(), JetID: msg.JetID, PulseNum: msg.PulseNum}, nil
 	}
 
-	pulse, err := h.PulseAccessor.Latest(ctx)
-	if err != nil {
-		return &reply.HeavyError{Message: err.Error(), JetID: msg.JetID, PulseNum: msg.PulseNum}, nil
-	}
-	futurePulse := pulse.NextPulseNumber
 	drop, err := storeDrop(ctx, h.DropModifier, msg.Drop)
 	if err != nil {
 		logger.Error(errors.Wrapf(err, "failed to store drop"))
 		return &reply.HeavyError{Message: err.Error(), JetID: msg.JetID, PulseNum: msg.PulseNum}, nil
 	}
-	if drop.Split {
-		_, _, err = h.JetModifier.Split(ctx, futurePulse, drop.JetID)
+	pulse := drop.Pulse
 
-	} else {
-		err = h.JetModifier.Update(ctx, futurePulse, false, drop.JetID)
+	// We can't to use PulseCalculator here because split pulse may exceed latest pulse.
+	whenSplit := pulse + 20
```

Comment:
What is 20 here? If it is 2 next pulses - it is wrong, because pulse length can be changed on pulsar

Reply:
Of course, I know and I wrote it just to make it clear and could be discussed. Any suggestions on how to get around?

Reply:
This is not correct. We should never assume the number of two pulses ahead. We can only assume that NextPulseNumber is correct.

Reply:
We need to fetch pulse from drop and take NextPulseNumber from it. It should be the split pulse, not two pulses ahead.

Reply:
Seems like jet.Split actually performed on the second pulse after drop.Pulse: https://github.com/insolar/insolar/blob/57ef5bbd8ba11be962a8003adca18cd2d7324686/ledger/light/executor/jet_splitter.go#L116